opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/CohortManager.java

```diff
@@ -751,6 +751,28 @@ private MyResourceId commonGetAnnotationSet(String id, @Nullable String studyStr
         return cohortAclList;
     }
 
+    public List<QueryResult<CohortAclEntry>> getAcls(String studyStr, List<String> cohortList, boolean silent,
+                                                     String sessionId) throws CatalogException {
+        MyResourceIds resource = getIds(cohortList, studyStr, sessionId);
+
+        List<QueryResult<CohortAclEntry>> cohortAclList = new ArrayList<>(resource.getResourceIds().size());
+        for (Long cohortId : resource.getResourceIds()) {
+            try {
+                QueryResult<CohortAclEntry> allCohortAcls =
+                        authorizationManager.getAllCohortAcls(resource.getStudyId(), cohortId, resource.getUser());
+                allCohortAcls.setId(String.valueOf(cohortId));
+                cohortAclList.add(allCohortAcls);
+            } catch (CatalogException e) {
+                if (silent) {
+                    cohortAclList.add(new QueryResult<>(studyStr, 0, 0, 0, "", e.toString(), new ArrayList<>(0)));
```

Comment:
The id of the queryResult should  not be studyStr but the corresponding element from cohortList